### ISSUE TITLE
Type alias created with `TypeAliasType` should generate error if referenced TypeVar is out of scope

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#6835]
This code should generate a type error but does not.

```python
S = TypeVar("S")

BadAlias2 = TypeAliasType("BadAlias2", list[S])  # Type error: S not in scope
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.344, which I just published. It will also be included in a future release of pylance.
